### PR TITLE
Create signup buttons

### LINKED ISSUE #42: Implement social sign-up buttons and functionality
**Task title**
Implement Social Sign-up Buttons and Functionality

**Task description**
Create components for social sign-up buttons, including Google, Facebook, and Twitter, and integrate the functionality to allow users to sign up using their social media accounts. This feature will enhance the user experience and streamline the registration process.

**Subtasks**

- [x] Design and create individual button components for Google, Facebook, and Twitter.

- [x] Implement authentication and authorization flows for each social media platform.

- [x] Add the necessary code to handle user registration through social sign-up.

- [ ] Test the functionality to ensure that users can successfully sign up using their social media accounts.

**Screenshots**
image

**Link to the component on Figma**
[Figma Link](https://www.figma.com/file/1ivLCv0Ardin5Ii83yuQp5/Student-Store-(Capstone)?type=design&node-id=2465-5867&mode=design&t=TpMh3OINCfQ9p7FC-4)

### LINKED ISSUE #43: Fix Firebase Auth Error (auth/invalid-api-key)
**Task title**
Fix Firebase Auth Error (auth/invalid-API-key)
**Task description**
We are encountering the "Firebase Auth Error (auth/invalid-API-key)" in our application, and we need to investigate and resolve this issue. This error is preventing the snapshot tests related to the Firebase from passing and the proper initialization of Firebase authentication and causing authentication-related issues. We need to ensure that Firebase is correctly configured and that the API key is valid.

**Screenshots**
image
image

### PR DESCRIPTION
**Pull request by Ben-Tewfik**
**Description:**

This pull request addresses two issues: "Implement Social Sign-up Buttons and Functionality" and "Fix Firebase Auth Error (auth/invalid-API-key)." I have completed the following tasks to resolve these issues:

**For the "Implement Social Sign-up Buttons and Functionality" issue:**

- Designed and created individual button components for Google, Facebook, and Twitter.
- Implemented authentication and authorization flows for each social media platform.
- Added the necessary code to handle user registration through social sign-up.

**For the "Fix Firebase Auth Error (auth/invalid-API-key)" issue:**
- Reviewed the Firebase configuration in the application to ensure that all credentials, including the API key, are correct.
- Validated Firebase initialization to ensure it's happening properly in the application.
- Mocked Firebase services in the test environment to prevent real API calls during testing. This ensures that Firebase initialization doesn't trigger actual requests with invalid credentials, thus resolving the "Firebase Auth Error (auth/invalid-API-key)".
- Verified that the testing environment is properly configured to use the correct Firebase project and credentials.

**Screenshots:**

- Implement Social Sign-up Buttons and Functionality

image

- Fix Firebase Auth Error (auth/invalid-API-key)

image

I request a review of this pull request to ensure the quality and correctness of the changes.

**Issue References:**

Closes #42 , Closes #43 

# Related Issue

- Resolve #42

### CODE REVIEW

File: src/components/FacebookButton/__test__/FacebookButton.test.js
Comment (hocine1212) [inline src/components/FacebookButton/__test__/FacebookButton.test.js:1]: Your implementation of this function has saved us a significant amount of time and spared us from unnecessary headaches.

**Review by khaoulasara — APPROVED** [review]
Impressive work on resolving the Firebase Auth issue swiftly and efficiently. The addition of social sign-up functionality is a great enhancement. Keep up the good work! 👏🔥

File: src/components/FacebookButton/FacebookButton.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { auth } from "@/util/firebase";
+import { FacebookAuthProvider, signInWithPopup } from "firebase/auth";
+import { BsFacebook } from "react-icons/bs";
+
+export default function FacebookButton({ children }) {
+    function signInWithFacebook() {
+        const provider = new FacebookAuthProvider();
+        signInWithPopup(auth, provider)
+            .then((result) => {
+                console.log(result);
```
Comment (hocine1212) [inline src/components/FacebookButton/FacebookButton.jsx:10]: Don't forget to get rid of the console logs later.

**Review by MouniaBelkheir — APPROVED** [review]
Well done and thanks for your hard work

**Review by hocine1212 — APPROVED** [review]
I really like the detailed PR and how you managed to solve the Firebase auth error, you did a great job so Keep up the good work .

File: src/components/FacebookButton/FacebookButton.jsx (hunk 1 above)
Comment (Ben-Tewfik) [reply src/components/FacebookButton/FacebookButton.jsx:10]: thank you for your feedback i will git rid of them.

File: src/components/FacebookButton/__test__/FacebookButton.test.js
Comment (Ben-Tewfik) [reply src/components/FacebookButton/__test__/FacebookButton.test.js:1]: thank you

**Pull request merged**